docker/div2k_train.py

```diff
@@ -0,0 +1,61 @@
+import os
+import matplotlib.pyplot as plt
+
+from data import DIV2K
+from model.srgan import generator, discriminator
+from train import SrganTrainer, SrganGeneratorTrainer
+
+# %matplotlib inline
+
+# Location of model weights (needed for demo)
+weights_dir = '/data/weights/div2k'
+os.makedirs(weights_dir, exist_ok=True)
+weights_file = lambda filename: os.path.join(weights_dir, filename)
+
+# Set up images
+images_dir = '/data/images'
+os.makedirs(images_dir, exist_ok=True)
+caches_dir = '/data/caches'
+os.makedirs(caches_dir, exist_ok=True)
+div2k_train = DIV2K(
+                    scale=4,
+                    subset='train',
+                    downgrade='bicubic',
+                    images_dir=images_dir,
+                    caches_dir=caches_dir,
+)
+div2k_valid = DIV2K(
+                    scale=4,
+                    subset='valid',
+                    downgrade='bicubic',
+                    images_dir=images_dir,
+                    caches_dir=caches_dir,
+)
+
+# Do pre-training
+check_dir = f'/data/ckpt/pre_generator'
+os.makedirs(check_dir, exist_ok=True)
+train_ds = div2k_train.dataset(batch_size=16, random_transform=True)
+valid_ds = div2k_valid.dataset(batch_size=16, random_transform=True, repeat_count=1)
+pre_trainer = SrganGeneratorTrainer(model=generator(), checkpoint_dir=check_dir)
+pre_trainer.train(
+                    train_ds,
+                    valid_ds.take(1),
+                    steps=1000000, 
+                    #steps=1000, 
+                    evaluate_every=1000, 
+                    save_best_only=False
+)
```

Comment:
I found that using EDSR instead of the original SRGAN generator gives better results after GAN training. See [this section](https://github.com/krasserm/super-resolution#srgan-for-fine-tuning-edsr-and-wdsr-models) for further details.

Reply:
Wait.. that looks like you're doing EDSR then SRGAN, vs first.. Sorry, this is the stuff I'm new to :)

Reply:
What I'm proposing is to use EDSR as generator instead of the generator from the SRGAN paper. The overall SRGAN training procedure remains the same. In other words, use `EdsrTrainer` (as described [here](https://github.com/krasserm/super-resolution#edsr-1)) instead of `SrganGeneratorTrainer`.

Reply:
Ahhh and done.